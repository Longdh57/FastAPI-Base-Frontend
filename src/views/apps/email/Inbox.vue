<template>
  <div>
    <MailToolbar/>
    <ul class="c-messages">
      <InboxMessage/>
      <InboxMessage read/>
      <InboxMessage read/>
      <InboxMessage/>
      <InboxMessage read/>
      <InboxMessage/>
      <InboxMessage read/>
      <InboxMessage read/>
      <InboxMessage read/>
      <InboxMessage read/>
      <InboxMessage read/>
      <InboxMessage/>
      <InboxMessage read/>
    </ul>
  </div>
</template>

<script>
import InboxMessage from './InboxMessage'
import MailToolbar from './MailToolbar'

export default {
  name: 'Inbox',
  components: {
    InboxMessage,
    MailToolbar
  }
}
</script>
