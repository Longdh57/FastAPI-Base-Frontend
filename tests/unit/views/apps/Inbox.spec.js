import {shallowMount, createLocalVue} from '@vue/test-utils';
import VueRouter from 'vue-router'
import CoreuiVue from '@coreui/vue-pro'
import Inbox from '@/views/apps/email/Inbox'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(CoreuiVue)

describe('Inbox.vue', () => {
  it('has a name', () => {
    expect(Inbox.name).toMatch('Inbox')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Inbox, {
      localVue,
      router
    })
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Inbox', () => {
    const wrapper = shallowMount(Inbox, {
      localVue,
      router
    })
    expect(wrapper.findComponent(Inbox)).toBeTruthy()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Inbox, {
      localVue,
      router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
