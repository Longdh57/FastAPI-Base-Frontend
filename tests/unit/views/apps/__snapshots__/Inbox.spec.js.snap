// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inbox.vue renders correctly 1`] = `
<div>
  <mailtoolbar-stub />
   
  <ul
    class="c-messages"
  >
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      title="Lorem ipsum dolor sit amet."
    />
     
    <inboxmessage-stub
      date="Today, 3:47 PM"
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      from="Lukasz Holeczek"
      read="true"
      title="Lorem ipsum dolor sit amet."
    />
  </ul>
</div>
`;
